relayer/pkg/chainlink/txm/txm.go

```diff
@@ -362,56 +354,34 @@ func (txm *starktxm) confirmLoop() {
 				break
 			}
 
-			allUnconfirmedTxs := txm.accountStore.GetAllUnconfirmed()
-			for accountAddressStr, unconfirmedTxs := range allUnconfirmedTxs {
+			for _, accountAddressStr := range txm.accountStore.Accounts() {
 				accountAddress, err := new(felt.Felt).SetString(accountAddressStr)
 				// this should never occur because the acccount address string key was created from the account address felt.
 				if err != nil {
 					txm.lggr.Errorw("could not recreate account address felt", "accountAddress", accountAddressStr)
 					continue
 				}
-				for _, unconfirmedTx := range unconfirmedTxs {
-					hash := unconfirmedTx.Hash
-					f, err := starknetutils.HexToFelt(hash)
-					if err != nil {
-						txm.lggr.Errorw("invalid felt value", "hash", hash)
-						continue
-					}
-					response, err := client.Provider.GetTransactionStatus(ctx, f)
-
-					// tx can be rejected due to a nonce error. but we cannot know from the Starknet RPC directly  so we have to wait for
-					// a broadcasted tx to fail in order to fix the nonce errors
-
-					if err != nil {
-						txm.lggr.Errorw("failed to fetch transaction status", "hash", hash, "nonce", unconfirmedTx.Nonce, "error", err)
-						continue
-					}
-
-					finalityStatus := response.FinalityStatus
-					executionStatus := response.ExecutionStatus
-
-					// any finalityStatus other than received
-					if finalityStatus == starknetrpc.TxnStatus_Accepted_On_L1 || finalityStatus == starknetrpc.TxnStatus_Accepted_On_L2 || finalityStatus == starknetrpc.TxnStatus_Rejected {
-						txm.lggr.Debugw(fmt.Sprintf("tx confirmed: %s", finalityStatus), "hash", hash, "nonce", unconfirmedTx.Nonce, "finalityStatus", finalityStatus)
-						if err := txm.accountStore.GetTxStore(accountAddress).Confirm(unconfirmedTx.Nonce, hash); err != nil {
-							txm.lggr.Errorw("failed to confirm tx in TxStore", "hash", hash, "accountAddress", accountAddress, "error", err)
-						}
-					}
-
-					// currently, feeder client is only way to get rejected reason
-					if finalityStatus == starknetrpc.TxnStatus_Rejected {
-						// we assume that all rejected transactions results in a unused rejected nonce, so
-						// resync. see the comment at resyncNonce for more details.
-						if resyncErr := txm.resyncNonce(ctx, client, accountAddress); resyncErr != nil {
-							txm.lggr.Errorw("resync failed for rejected tx", "error", resyncErr)
-						}
-
-						go txm.logFeederError(ctx, hash, f)
-					}
-
-					if executionStatus == starknetrpc.TxnExecutionStatusREVERTED {
-						// TODO: get revert reason?
-						txm.lggr.Errorw("transaction reverted", "hash", hash)
+				nonce, err := client.AccountNonceLatest(ctx, accountAddress)
```

Comment:
I thought that this was going to supplement the existing confirmation logic. any reason why you wanted to remove it? 

It's much cleaner this way, but we should make a note this will need to be rolled out slowly. it'd actually be helpful to introduce a feature flag service for NOPs for cases like this.

Reply:
That was the initial goal, but confirming transactions individually proved to be much more unreliable than we thought. `GetTransactionStatus` is very slow, and almost every time it will return an `unknown transaction` error when fetching an already broadcasted transaction. This not only doesn't provide any benefits over the new design, but it can potentially generate a lot of false positives for freshly broadcast transactions.

Re feature flag, I don't have any strong preference.